src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -833,6 +886,38 @@ class PersonAgent(
           } else {
             currentVehicle
           }
+        // todo rrp
+        currentVehicleForNextState.headOption.foreach { beamVehicleId =>
+          import beamScenario.beamConfig.beam.agentsim.agents.vehicles.{
+            noRechargeThresholdInMeters,
+            rechargeRequiredThresholdInMeters
+          }
+
+          val vehicle = beamVehicles(beamVehicleId).vehicle
+          if (vehicle.isBEV || vehicle.isPHEV) {
+            val distanceToDestinationInMeters = {
+              trip.view
+                .takeWhile(_.beamVehicleId == vehicle.id)
+                .foldLeft(0.0)(_ + _.beamLeg.travelPath.distanceInM)
+            }
+            val newRechargeRequiredThresholdInMeters = {
+              rechargeRequiredThresholdInMeters +
+              (1 - rechargeRequiredThresholdInMeters / distanceToDestinationInMeters) *
+              distanceToDestinationInMeters
+            }
+            // todo: compute new noRechargeThresholdInMeters using distanceToDestinationInMeters value.
+            val newNoRechargeThresholdInMeters = noRechargeThresholdInMeters
+            val refuelNeeded =
+              vehicle.isRefuelNeeded(newRechargeRequiredThresholdInMeters, newNoRechargeThresholdInMeters)
+
+            if (refuelNeeded) {
+              val enrouteCharging = EnrouteCharging(
+                destinationUTM = beamServices.geo.wgs2Utm(trip.last.beamLeg.travelPath.endPoint.loc)
```

Comment:
@dimaopen, is this the correct way to find destination of a trip? We want to remember this to re-route the vehicle once it done recharging.

Reply:
@dimaopen, made the change. See [here](https://github.com/LBNL-UCB-STI/beam/pull/3257/files#diff-940cd21d65d4ec3f2f9c49c5aaf9753e3b2b4d030fabd7a7d4ed861a8eda857fR898), we basically use it for two things.
1) to find distance 
2) to find end point